internal/rest/dequeue_nodes.go

```diff
@@ -699,7 +699,15 @@ func (rest *RestOperations) ExecuteRestAndPopulateCache(rest_ops []*utils.RestOp
 
 func updateGatewayStatusWithVsError(key string, rest_op *utils.RestOp) {
 	if rest_op.Model == "VirtualService" {
-		vs := rest_op.Obj.(avimodels.VirtualService)
+		var vs avimodels.VirtualService
+		switch rest_op.Obj.(type) {
+		case avimodels.VirtualService:
+			vs = rest_op.Obj.(avimodels.VirtualService)
+		case *avimodels.VirtualService:
```

Comment:
As logic at line number 711 is for Parent VS only, should we return from line number 707? no need to check type for child vs.. right?

Reply:
let's keep it as defensive check in case there is unexplored scenario for parentvs  where type can be pointer.